compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5922,7 +5922,7 @@ public void visit(BLangSimpleVarRef varRefExpr) {
                 if (constSymbol.literalType.tag <= TypeTags.BOOLEAN || constSymbol.literalType.tag == TypeTags.NIL) {
                     BLangLiteral literal = ASTBuilderUtil.createLiteral(varRefExpr.pos, constSymbol.literalType,
                             constSymbol.value.value);
-                    result = rewriteExpr(addConversionExprIfRequired(literal, varRefExpr.getBType()));
+                    result = addConversionExprIfRequired(literal, varRefExpr.getBType());
```

Comment:
Do we need this change? everything works fine with `rewriteExpr`

Reply:
IMO, it is a kind of an unnecessary call since we change the implicit casts into expressions within the `addConversionExprIfRequired`. Is there any other specific reason to call `rewriteExpr` here?

Reply:
Usually we call this for desugared call. Better to keep it.

Reply:
Ah okay. Will update.